pkg/k8shandler/common.go

```diff
@@ -181,15 +181,30 @@ func newElasticsearchContainer(imageName string, envVars []v1.EnvVar, resourceRe
 	}
 }
 
-func newProxyContainer(imageName, clusterName, namespace string, logConfig LogConfig) (v1.Container, error) {
+func newProxyContainer(spec api.ProxySpec, imageName, clusterName, namespace string, logConfig LogConfig) (v1.Container, error) {
 	cpuLimit, err := resource.ParseQuantity("100m")
```

Comment:
we have a function that will handle resource requirements, can we leverage that instead? (see `newPodTemplateSpec` for reference)